src/mime.cr

```diff
@@ -0,0 +1,167 @@
+require "crystal/system/mime"
+
+# This module implements a global MIME registry.
+#
+# ```
+# require "mime"
+# MIME[".html"] # => "text/html"
+# ```
+#
+# The registry is populated with some defaults values (see `DEFAULT_TYPES`)
+# as well as the operating system's MIME database.
+#
+# Applications can register their own MIME types:
+#
+# ```
+# MIME.fetch?(".cr") # => nil
+# MIME.extensions("text/crystal") # => []
+
+# MIME.register(".cr", "text/crystal")
+# MIME.fetch?(".cr") # => "text/crystal"
+# MIME.extensions("text/crystal") # => [".cr"]
+# ```
+module MIME
+  class Error < Exception
+  end
+
+  @@initialized = false
+  @@types = {} of String => String
+  @@types_lower = {} of String => String
+  @@extensions = {} of String => Array(String)
+
+  # A limited set of default MIME types.
+  DEFAULT_TYPES = {
+    ".css"  => "text/css; charset=utf-8",
+    ".gif"  => "image/gif",
+    ".htm"  => "text/html; charset=utf-8",
+    ".html" => "text/html; charset=utf-8",
+    ".jpg"  => "image/jpeg",
```

Comment:
It might be worth mapping `.jpeg` here too, since that gets used pretty often in the wild.

Reply:
I'd rather keep this list at a very minimum. These extensions are the same that the Go implementation provides by default plus `.txt` which was previously supported by `HTTP::StaticFileHandler`.

Reply:
Makes sense, but consider cURL's [`mime.c`](https://github.com/curl/curl/blob/master/lib/mime.c#L1653):

```c
static const struct ContentType ctts[] = {
    {".gif",  "image/gif"},
    {".jpg",  "image/jpeg"},
    {".jpeg", "image/jpeg"},
    {".png",  "image/png"},
    {".svg",  "image/svg+xml"},
    {".txt",  "text/plain"},
    {".htm",  "text/html"},
    {".html", "text/html"},
    {".pdf",  "application/pdf"},
    {".xml",  "application/xml"}
};
```

`.jpeg` is about as common as it gets in terms of alternative suffixes, and it's only one additional line.